### PR TITLE
Network: small memory optimization

### PR DESCRIPTION
**Pull request by tbursztyka**

### CODE REVIEW

File: subsys/net/l2/ethernet/arp.c [hunk 1]
```diff
@@ -179,14 +179,14 @@ static void arp_entry_register_pending(struct arp_entry *entry)
...
-	s64_t current = k_uptime_get();
+	u32_t current = k_uptime_get_32();
 	struct arp_entry *entry, *next;
 
 	ARG_UNUSED(work);
 
 	SYS_SLIST_FOR_EACH_CONTAINER_SAFE(&arp_pending_entries,
 					  entry, next, node) {
-		if ((entry->req_start + ARP_REQUEST_TIMEOUT - current) > 0) {
+		if (entry->req_start + ARP_REQUEST_TIMEOUT > current) {
```
Comment (jukkar) [inline subsys/net/l2/ethernet/arp.c]: Will this work properly if req_start + ARP_REQ_TIMEOUT wraps over?
We could do it in similar way as how `include/net/net_ip.h:net_tcp_seq_greater()` does it.

**Review by jukkar — APPROVED** [review]

**Comment by pfalcon** [comment]
> However, since the cove to k_work queues (due to the introduction of TC)

Failed to parse this.

**Comment by pfalcon** [comment]
> net/pkt: Remove useless _reserved attribute
> This attribute was used when net_pkt where queued into a fifo.

net_pkt's *are* queued into fifo, e.g. in sockets. This looks like breaking change.

**Comment by tbursztyka** [comment]
> net_pkt's are queued into fifo, e.g. in sockets. This looks like breaking change.

Yep missed that. However, it's on receiving side only: at this point the k_fifo_put is using the k_work area which is then useless (since to get to socket, the k_work got consumed). 
Thus why it did not break anything. (k_fifo only need 4 bytes and k_work is 12 bytes I believe).

Anyway, I'll send a patch to make this clearer in net_pkt.h

**Comment by pfalcon** [comment]
@tbursztyka 

> However, it's on receiving side only: at this point the k_fifo_put is using the k_work area which is then useless

From quick look I had into that, I had a similar hunch. Thanks for looking into that!

**Pull request merged**